Mage/src/main/java/mage/util/CardUtil.java

```diff
@@ -1751,9 +1752,9 @@ public static Map<String, Object> getSourceCostsTagsMap(Game game, Ability sourc
         }
 
         // from any ability before resolve (on stack) - access by spell ability
-        MageObject sourceObject = source.getSourceObject(game);
```

Comment:
> from any ability before resolve (on stack) - access by spell ability

You keep comment that it called before resolve, but now it uses lki. That’s wrong logic. Must keep old logic or remove wrong comment (with your new code it can be called after spell removed from stack).

Reply:
Good point, will fix comment.